index.html

```diff
@@ -3549,28 +3548,40 @@ <h3><dfn>Writing or pushing content</dfn></h3>
         the <a>activated reader objects</a>, run the following sub-steps:
         <ol>
           <li>
-            If |reader|.<a>[[\Id]]</a> is [= dictionary member/present =]
-            and it is not equal to any |record|'s
-            <a href="#dom-ndefrecord-id">id</a> where |record|
-            is an element of |message|, [= iteration/continue =].
-          </li>
-          <li>
-            If |reader|.<a>[[\RecordType]]</a> is [= dictionary member/present =]
-            and it is not equal to any |record|'s
-            <a href="#dom-ndefrecord-recordtype">recordType</a> where |record|
-            is an element of |message|, [= iteration/continue =].
-          </li>
-          <li>
-            If |reader|.<a>[[\MediaType]]</a> is not `""` and
-            it is not equal to any |record|'s
-            <a href="#dom-ndefrecord-mediatype">mediaType</a> where |record| is
-            an element of |message|, [= iteration/continue =].
-          </li>
-          <li>
-            <a>Fire an event</a> named "`reading`" at |reader|
-            using <a>NDEFReadingEvent</a> with its <a>serialNumber</a> attribute
-            initialized to |serialNumber| and <a>message</a> attribute
-            initialized to |message|.
+            [= list/For each =] |record:NDEFRecord| in |message:NDEFMessage|,
+            <ol>
+              <li>
+                Let |matched| be `undefined`.
+              </li>
+              <li>
+                If |reader|.<a>[[\Id]]</a> is not `""` and is not equal to
+                |record|'s <a href="#dom-ndefrecord-id">id</a>,
+                [= iteration/continue =] with the next record.
+                If it is equal, set |matched| to `true`.
+              </li>
+              <li>
+                If |reader|.<a>[[\RecordType]]</a> is
+                [= dictionary member/present =] and it is not equal to
+                |record|'s <a href="#dom-ndefrecord-recordtype">recordType</a>,
+                [= iteration/continue =] with the next record.
+                If it is equal and if |matched| is `undefined`, set |matched|
+                to `true`.
+              </li>
+              <li>
+                If |reader|.<a>[[\MediaType]]</a> is not `""` and is not equal
+                to |record|'s <a href="#dom-ndefrecord-mediatype">mediaType</a>,
+                [= iteration/continue =] with the next record.
+                If it is equal and if |matched| is `undefined`, set |matched|
+                to `true`.
+              </li>
+              <li>
+                If |matched| is `true`, <a>fire an event</a> named "`reading`"
+                at |reader| using <a>NDEFReadingEvent</a> with its
+                <a>serialNumber</a> attribute initialized to |serialNumber| and
+                <a>message</a> attribute initialized to |message| and
+                [= iteration/continue =] with the next reader.
```

Comment:
It is weird that you do continue with another loop (reader?) here..

Reply:
We should use "break" as @Honry suggested since we want to continue with the outer loop (next reader).